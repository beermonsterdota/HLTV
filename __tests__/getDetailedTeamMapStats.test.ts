import HLTV from '../src/index.js'
import { sleep } from '../src/utils.js'
import { expect, test } from 'vitest'

const FULL = 9863
const EMPTY = 9918

test('getDetailedTeamMapStats', async () => {
  await sleep(3000)
  expect(await HLTV.getDetailedTeamMapStats({ id: FULL })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getDetailedTeamMapStats({ id: EMPTY })).toMatchSnapshot()
  await sleep(3000)
}, 30000)
