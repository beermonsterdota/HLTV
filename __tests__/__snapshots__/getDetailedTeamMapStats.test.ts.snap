// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getDetailedTeamMapStats 1`] = `
{
  "de_ancient": {
    "ctRoundWinPercent": 61.7,
    "draws": 0,
    "losses": 3,
    "pistolRoundWinPercent": 40,
    "pistolRounds": "10",
    "pistolRoundsWon": "4",
    "roundsWon": "63",
    "tRoundWinPercent": 36.1,
    "timesPlayed": "5",
    "totalRounds": "132",
    "winPercent": 40,
    "wins": 2,
  },
  "de_cache": {
    "ctRoundWinPercent": 28.6,
    "draws": 0,
    "losses": 1,
    "pistolRoundWinPercent": 50,
    "pistolRounds": "2",
    "pistolRoundsWon": "1",
    "roundsWon": "13",
    "tRoundWinPercent": 60,
    "timesPlayed": "1",
    "totalRounds": "29",
    "winPercent": 0,
    "wins": 0,
  },
  "de_dust2": {
    "ctRoundWinPercent": 46.6,
    "draws": 0,
    "losses": 42,
    "pistolRoundWinPercent": 48,
    "pistolRounds": "150",
    "pistolRoundsWon": "72",
    "roundsWon": "974",
    "tRoundWinPercent": 48.7,
    "timesPlayed": "75",
    "totalRounds": "2044",
    "winPercent": 44,
    "wins": 33,
  },
  "de_inferno": {
    "ctRoundWinPercent": 47.4,
    "draws": 0,
    "losses": 60,
    "pistolRoundWinPercent": 42.3,
    "pistolRounds": "208",
    "pistolRoundsWon": "88",
    "roundsWon": "1287",
    "tRoundWinPercent": 47.6,
    "timesPlayed": "104",
    "totalRounds": "2711",
    "winPercent": 42.3,
    "wins": 44,
  },
  "de_mirage": {
    "ctRoundWinPercent": 54.4,
    "draws": 0,
    "losses": 42,
    "pistolRoundWinPercent": 44.7,
    "pistolRounds": "152",
    "pistolRoundsWon": "68",
    "roundsWon": "992",
    "tRoundWinPercent": 41.6,
    "timesPlayed": "76",
    "totalRounds": "2062",
    "winPercent": 44.7,
    "wins": 34,
  },
  "de_nuke": {
    "ctRoundWinPercent": 43.3,
    "draws": 0,
    "losses": 10,
    "pistolRoundWinPercent": 40,
    "pistolRounds": "30",
    "pistolRoundsWon": "12",
    "roundsWon": "166",
    "tRoundWinPercent": 42.9,
    "timesPlayed": "15",
    "totalRounds": "385",
    "winPercent": 33.3,
    "wins": 5,
  },
  "de_overpass": {
    "ctRoundWinPercent": 52.7,
    "draws": 0,
    "losses": 40,
    "pistolRoundWinPercent": 48.3,
    "pistolRounds": "174",
    "pistolRoundsWon": "84",
    "roundsWon": "1145",
    "tRoundWinPercent": 48.1,
    "timesPlayed": "87",
    "totalRounds": "2274",
    "winPercent": 54,
    "wins": 47,
  },
  "de_train": {
    "ctRoundWinPercent": 52.5,
    "draws": 0,
    "losses": 42,
    "pistolRoundWinPercent": 47.3,
    "pistolRounds": "188",
    "pistolRoundsWon": "89",
    "roundsWon": "1270",
    "tRoundWinPercent": 49,
    "timesPlayed": "94",
    "totalRounds": "2506",
    "winPercent": 55.3,
    "wins": 52,
  },
  "de_vertigo": {
    "ctRoundWinPercent": 45.4,
    "draws": 0,
    "losses": 26,
    "pistolRoundWinPercent": 47.9,
    "pistolRounds": "94",
    "pistolRoundsWon": "45",
    "roundsWon": "597",
    "tRoundWinPercent": 49.7,
    "timesPlayed": "47",
    "totalRounds": "1260",
    "winPercent": 44.7,
    "wins": 21,
  },
}
`;

exports[`getDetailedTeamMapStats 2`] = `{}`;
